### DISCUSSION TITLE
React Native

### DISCUSSION THREAD

**Discussion by jeanrobatto** [outfoxx/Shield#30] [General]
Hey! 

Anyone know how to link this library to a React Native custom module and get it to build?

I keep seeing the "No such module: Shield" error.

Thanks in advance!

**Comment by kdubb** [comment]
I personally don't have any experience with React Native. Shield a pretty _simple_ SPM module, although it does have other dependencies.

What version of Xcode are you using to build your module?

**Reply by jeanrobatto** [reply]
thank you for your quick reply! I'm using Xcode 13.1

Essentially what I'm trying to do is to create a native module for React Native and adding Shield as a dependency.

With Cocoapods, all I have to do is add the dependency in the .podspec file and voila

But i'm not familiar with the SwiftPM, so now I'm trying to add Shied as a package dependency to my Pods project within the RN workspace, then adding it manually as a dependency to my custom Pod target

unfortunately i still get the "no such module"

**Reply by jeanrobatto** [reply]
Honestly im not 100% familiar with swift/Xcode so maybe im getting the concept wrong

**Reply by kdubb** [reply]
I think this might be a question better suited for a React Native group. If they can help you add a Swift package to your project. I'll be here to help you with using Shield

**Reply by jeanrobatto** [reply]
great, thanks!
